File: src/main/java/com/hedvig/backoffice/services/account/dto/AccountEntryDTO.java
package com.hedvig.backoffice.services.account.dto;

import lombok.Value;

import javax.money.MonetaryAmount;
import java.time.Instant;
import java.time.LocalDate;
import java.util.Optional;
import java.util.UUID;

@Value
public class AccountEntryDTO {
  UUID id;
  LocalDate fromDate;
  MonetaryAmount amount;
  AccountEntryType type;
  String source;
  String reference;
  Optional<String> title;
  Optional<String> comment;
  String addedBy;
  Optional<Instant> failedAt;
  Optional<Instant> chargedAt;
}

File: src/main/java/com/hedvig/backoffice/services/account/dto/AccountEntryDTO.kt
package com.hedvig.backoffice.services.account.dto

import java.time.Instant
import java.time.LocalDate
import java.util.UUID
import javax.money.MonetaryAmount

data class AccountEntryDTO(
    val id: UUID,
    val fromDate: LocalDate,
    val amount: MonetaryAmount,
    val type: String,
    val source: String,
    val reference: String,
    val title: String?,
    val comment: String?,
    val addedBy: String,
    val failedAt: Instant?,
    val chargedAt: Instant?
)
